spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Dropout.scala

```diff
@@ -26,10 +26,14 @@ import scala.reflect.ClassTag
 
 @SerialVersionUID(- 4636332259181125718L)
 class Dropout[T: ClassTag](
-  val initP: Double = 0.5, val inplace: Boolean = false, var scale: Boolean = true)(
+  val initP: Double = 0.5,
+  val inplace: Boolean = false,
+  var scale: Boolean = true,
+  val isLazy: Boolean = false)(
   implicit ev: TensorNumeric[T]) extends TensorModule[T] {
   private var p = initP
-  val noise = Tensor[T]()
+  var noise = Tensor[T]()
+  var flag = Array(true) // used by lazy noise
```

Comment:
Because, the flag need to be shared by dropouts in each time step.
A boolean is a primitive value not being able to be shared. So I used a array to make dropouts point to the same reference.